### PR TITLE
Twt sleep status fix

### PR DESCRIPTION
**Pull request by ajayparida**

### CODE REVIEW

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c [hunk 1]
```diff
@@ -822,13 +822,17 @@ void wifi_nrf_event_proc_twt_sleep_zep(void *vif_ctx,
 
 		fmac_dev_ctx->twt_sleep_status = WIFI_NRF_FMAC_TWT_STATE_SLEEP;
 
+		wifi_mgmt_raise_twt_sleep_state(vif_ctx_zep->zep_net_if_ctx,
```
[the remark this answers is not in the document]
Comment (krish2718) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_mgmt.c:825]: I have raised an internal Jira to discuss whether to queue it in the driver or let application handle, but as this is working, resolving this comment.

**Review by krish2718 — APPROVED** [review]

**Review by sachinthegreen — CHANGES_REQUESTED** [review]
I guess this needs a Changelog update, since it introduces a new capability?

@kittydepa : FYI.

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 2]
```diff
@@ -354,6 +354,9 @@ DTIM-based power save performs better in high throughput applications compared t
 As the device sleeps longer and does not wake up to receive DTIM beacons, it misses all multicast or broadcast frames.
 The TWT session is expected to be set up by the application after the network level negotiation, after which it is not expected to receive any multicast or broadcast frames.
 
+Additionally, by using the NET_EVENT_WIFI_TWT_SLEEP_STATE, the application layer can be configured to subscribe to the TWT events
```
Comment (kittydepa) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:357]:
```suggestion
Additionally, by using the ``NET_EVENT_WIFI_TWT_SLEEP_STATE``, the application layer can be configured to subscribe to the TWT events
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 3]
```diff
@@ -414,6 +417,10 @@ The following shell commands and network management APIs are provided for power
      - wifi twt teardown_all
      - Tear down all sessions
      - TWT operation TWT teardown all flows
+   * - net_mgmt(NET_EVENT_WIFI_TWT_SLEEP_STATE)
+     -
+     - Event,application need to register for retrieving TWT sleep state.
```
Comment (kittydepa) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:422]:
```suggestion
     - Event application needed to register for retrieving TWT sleep state.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 4]
```diff
@@ -414,6 +417,10 @@ The following shell commands and network management APIs are provided for power
...
+     - Apllication will be notified with the event along with TWT sleep state info.
```
Comment (kittydepa) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:423]:
```suggestion
     - Application will be notified with the event along with TWT sleep state information.
```

**Review by kittydepa — APPROVED** [review]

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 5]
```diff
@@ -354,6 +354,9 @@ DTIM-based power save performs better in high throughput applications compared t
 As the device sleeps longer and does not wake up to receive DTIM beacons, it misses all multicast or broadcast frames.
 The TWT session is expected to be set up by the application after the network level negotiation, after which it is not expected to receive any multicast or broadcast frames.
 
+Additionally, by using the ``NET_EVENT_WIFI_TWT_SLEEP_STATE``, the application layer can be configured to subscribe to the TWT events
```
Comment (sachinthegreen) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:357]:
```suggestion
Additionally, by using the ``NET_EVENT_WIFI_TWT_SLEEP_STATE``, the application layer can subscribe to TWT events
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 6]
```diff
@@ -354,6 +354,9 @@ DTIM-based power save performs better in high throughput applications compared t
...
+so that the application knows if it is the service period, and the time to send or receive data.
```
Comment (sachinthegreen) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:358]:
```suggestion
to be notified of beginning and end of service period, so as to be able to send or receive data.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 7]
```diff
@@ -414,6 +417,10 @@ The following shell commands and network management APIs are provided for power
      - wifi twt teardown_all
      - Tear down all sessions
      - TWT operation TWT teardown all flows
+   * - net_mgmt(NET_EVENT_WIFI_TWT_SLEEP_STATE)
```
Comment (sachinthegreen) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:420]: Shouldn't this be net_mgmt_event_notify_with_info(NET_EVENT_WIFI_TWT_SLEEP_STATE) ?

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 8]
```diff
@@ -414,6 +417,10 @@ The following shell commands and network management APIs are provided for power
...
+     -
```
Comment (sachinthegreen) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:421]: Put a N/A here. Also maybe change the name of the column to `Wi-Fi shell command` ?

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Review by sachinthegreen — COMMENTED** [review]
@udaynordic : Can you please review this too?

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 9]
```diff
@@ -414,6 +417,10 @@ The following shell commands and network management APIs are provided for power
...
+     - Event application needed to register for retrieving TWT sleep state.
```
Comment (sachinthegreen) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:422]:
```suggestion
     - Applications can register to this event to be notified about TWT sleep/wake events.
```

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst [hunk 10]
```diff
@@ -414,6 +417,10 @@ The following shell commands and network management APIs are provided for power
...
+     - Application will be notified with the event along with TWT sleep state information.
```
Comment (sachinthegreen) [inline doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:423]: This statement can be removed if we make the above change.

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst (hunk 8 above)
Comment (kittydepa) [reply doc/nrf/device_guides/working_with_nrf/nrf70/developing/powersave.rst:421]: If so, please fill in the **Command** column:
image


with the text: wifi shell 
Unless, the word "command" is part of the command.

**Review by sachinthegreen — APPROVED** [review]

**Pull request merged**